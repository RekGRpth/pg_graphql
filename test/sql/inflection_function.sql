begin;
    create view f as
        select
            distinct name
        from
            graphql.field
        where
            func is not null
        order by
            name;

    create table account (
        id int primary key
    );

    create function _full_name(rec public.account)
        returns text
        immutable
        strict
        language sql
    as $$
        select 'Foo';
    $$;

    -- Inflection off, Overrides: off
    comment on schema public is e'@graphql({"inflect_names": false})';
    select * from f;

    savepoint a;

    -- Inflection off, Overrides: on
    comment on function public._full_name(public.account) is E'@graphql({"name": "wholeName"})';
    select * from f;

    rollback to savepoint a;

    -- Inflection on, Overrides: off
    comment on schema public is e'@graphql({"inflect_names": true})';
    select * from f;

    -- Inflection on, Overrides: on
    comment on function public._full_name(public.account) is E'@graphql({"name": "WholeName"})';
    select * from f;

rollback;
