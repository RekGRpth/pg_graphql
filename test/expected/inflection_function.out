begin;
    create view f as
        select
            distinct name
        from
            graphql.field
        where
            func is not null
        order by
            name;
    create table account (
        id int primary key
    );
    create function _full_name(rec public.account)
        returns text
        immutable
        strict
        language sql
    as $$
        select 'Foo';
    $$;
    -- Inflection off, Overrides: off
    comment on schema public is e'@graphql({"inflect_names": false})';
    select graphql.rebuild_schema();
 rebuild_schema 
----------------
 
(1 row)

    select * from f;
   name    
-----------
 full_name
(1 row)

    savepoint a;
    -- Inflection off, Overrides: on
    comment on function public._full_name(public.account) is E'@graphql({"name": "wholeName"})';
    select graphql.rebuild_schema();
 rebuild_schema 
----------------
 
(1 row)

    select * from f;
   name    
-----------
 wholeName
(1 row)

    rollback to savepoint a;
    -- Inflection on, Overrides: off
    comment on schema public is e'@graphql({"inflect_names": true})';
    select graphql.rebuild_schema();
 rebuild_schema 
----------------
 
(1 row)

    select * from f;
   name   
----------
 fullName
(1 row)

    -- Inflection on, Overrides: on
    comment on function public._full_name(public.account) is E'@graphql({"name": "WholeName"})';
    select graphql.rebuild_schema();
 rebuild_schema 
----------------
 
(1 row)

    select * from f;
   name    
-----------
 WholeName
(1 row)

rollback;
